### DISCUSSION TITLE
How to iterate through a list of sockets at a specific namespace in a specific room?

### DISCUSSION THREAD

**Discussion by ghost** [socketio/socket.io#5130] [Q&A] [answered]
Below here is code from the private messaging example.
It returns all the sockets and is iterable so I can get all the ID's and the username of users connected to the server PERIOD.

My problem is I need to be able to do the below but only for a specific room, being the one that the user is connected to. How can I do this?
```js
    let users = [];
    for (let [id, socket] of io.of("/").sockets) {
      users.push({
        userID: id,
        username: socket.username,
        message: "",
      });
    }
```

**Comment by darrachequesne** [comment, marked as answer]
You can use the `fetchSockets()` methods:

```js
const sockets = await io.in("my-room").fetchSockets();
```

Reference: https://socket.io/docs/v4/server-api/#namespacefetchsockets
